### ISSUE TITLE
[rstream] common options for streams & subscription types

### ISSUE THREAD

**Issue by postspectacular** [thi-ng/umbrella#92]
(edited: added new type specific options to collect ideas in one place)

[As mentioned previously](https://github.com/thi-ng/umbrella/issues/81#issuecomment-482062843), I'd like to update all stream & subscription ctors (and their syntax-sugared factory functions) to accept an object of config options. Since there're several shared behaviors between the various subscription types, these options should be as unified as possible, with type specific additional options.

This issue exist to collect a list of the common aspects one might wish to configure, please feel free to add your thoughts. At the time of writing, I'm still in the middle of getting various thi.ng/webgl related packages ready for initial release, but would then be more free to address this in the near future.

Here's an initial, non-exhaustive list, names debatable:

- `id` - stream/subscription ID
- `xform` - incoming value transducer
- `keepAlive` - auto-termination behavior after last subscriber unsubbed (see #74, #81, #91)

Type specific options:

### sidechainPartition

- `lastOnly: boolean` - if true, only returns the last collected value

This will simplify this current pattern and avoid the `map(peek)` transducer:

```ts
foo.subscribe(sidechainPartition(fromRAF())).transform(map(peek))...
```

To:

```ts
foo.subscribe(sidechainPartition(fromRAF(), { lastOnly: true }))...
```

**Comment by gavinpc-mindgrub** [comment]
This makes sense to me, as I frequently have to reference the overload list.  This would also make the implementation more direct.

Just to make sure I'm understanding the intent...

I often use this overload:

```
foo.subscribe(tx.map(f));
```
With this proposal, would that now be written as

```
foo.subscribe({ xform: tx.map(f) });
```

Would these options be in addition to any subscriber object?

For example if you wanted to include an error trap, would it be
```
foo.subscribe({ error(e) { handle(error); } }, { xform: tx.map(f) });
```

i.e. there would still be some overloads.

**Comment by postspectacular** [comment]
Hi @gavinpc-mindgrub - the `xform` option is already largely present and yes, both approaches (i.e. what you illustrated) are possible/viable...

The options object in question here is what you'd pass to say `stream()`, `fromXXX()`, `sync()`, `merge()` etc. wasn't intending to change the params for `.subscribe()` itself...

**Comment by postspectacular** [comment]
Implemented since v3.0.0
